presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -995,7 +996,15 @@ public Node visitSpecialDateTimeFunction(SqlBaseParser.SpecialDateTimeFunctionCo
     @Override
     public Node visitExtract(SqlBaseParser.ExtractContext context)
     {
-        return new Extract(getLocation(context), (Expression) visit(context.valueExpression()), Extract.Field.valueOf(context.identifier().getText().toUpperCase()));
+        String fieldString = context.identifier().getText();
+        Extract.Field field = null;
+        try {
+            field = Extract.Field.valueOf(fieldString.toUpperCase());
+        }
+        catch (IllegalArgumentException e) {
+            check(field != null, format("Invalid EXTRACT field: %s", fieldString), context);
```

Comment:
@martint  Since I want to re-use `check` here and must have a condition here, is it okay to use `check(false, ...)` or I have to use `throw new ParsingException(...)` instead?